stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/messages/reactions/ReactionsView.kt

```diff
@@ -56,6 +56,15 @@ public abstract class ReactionsView : RecyclerView {
         reactionsAdapter.submitList(createReactionItems(message, isMyMessage))
     }
 
+    public fun setReaction(reaction: Reaction, isMyReaction: Boolean = false) {
+        // according to the design, current user reactions have the same style
+        // as reactions on the current user messages in the message list
+        this.isMyMessage = isMyReaction
+
+        setPadding(0, 0, 0, 0)
+        reactionsAdapter.submitList(listOf(ReactionItem(reaction, true)))
+    }
```

Comment:
Am I understanding it correctly that we're using a separate instance of this RecyclerView (`ReactionsView`) for every reaction shown in `UserReactionsView`?

Maybe we should extract `ReactionViewHolder`'s layout into a View, and use that inside `UserReactionsView`?

Reply:
We discussed this, and it would cause a lot of duplication, and the performance isn't necessarily an issue with the current implementation.